src/TropicalGeometry/matrix.jl

```diff
@@ -30,12 +32,9 @@ julia> det(A)
 ```
 """
 function det(A::Generic.MatSpaceElem{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
```

Comment:
This function is also being called for matrices over `MPolyRingElem{<:TropicalSemiringElem}` and `PolyRingElem{<:TropicalSemiringElem}`, which causes problems in the tests.

We probably want two `det` functions:
-  `function det(A::Generic.MatSpaceElem{TropicalSemiringElem})` (this function)
-  `function det(A::Generic.MatSpaceElem{R}) where {R<:Union{MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}` (the old `det` function)

Reply:
Yes, I saw that.  The following test fails
```
R, (x,y) = T[:x,:y]
A = Matrix(identity_matrix(R,2)) # (julia) matrix over tropical polynomials
@test det(A) == one(R)
```
I am not quite sure if your old was actually correct.  Of course, the Leibniz formula can be evaluated over an arbitrary semiring.  But I don't understand what it should mean here.

The reason why ordinary determinants of matrices with ordinary polynomials as coefficients make sense is that polynomial rings don't have zero divisors, whence they embed into their field of fractions.  Ordinary determinants only make sense over ordinary fields.

This situation looks very different to me.  So, please enlighten me about your goals here.

Reply:
Note: For coefficients in the tropical semiring there is no difference between the (tropical) determinant and the (tropical) permanent.  the reason is that +1 and -1 in any valued field both tropicalize to 0.

Reply:
I added determinants of matrices over the tropical polynomial ring without a second thought, and I don't really have any goals with them.

If determinants over the tropical polynomial ring don't make any sense mathematically, then I agree that we should remove those failing tests and possibly even change the tropical `det` function so that it is only called for matrices over the tropical semiring.

That way, if the user tries to compute `det` of a matrix of tropical polynomials, OSCAR will use the generic implementation and the user will immediately get an error that negation is not defined over the tropical semiring.

Reply:
OK, I removed those tests and the function definition for matrices with polynomials.